### DISCUSSION TITLE
Set frozen channel from client side

### DISCUSSION THREAD

**Discussion by tristan-garaud** [GetStream/stream-chat-android#4497] [Q&A]
Hi,
I want to allow my user to block a user from my android client. When I freeze my channel, both users can still send message to the channel. Base on the dock I was expecting to get an error 403 but everything looks fine. I don't know if there is something to do on the backend side or if there is a permission to set/remove (like UseFrozenChannel).

Thanks

**Comment by adasiewiczr** [comment]
Hey @tristan-garaud,
indeed, if you freeze the channel users are not able to send messages unless they have `UserFrozenChannel` permission. You can find more information [here](https://getstream.io/chat/docs/android/disabling_channels/?language=kotlin#freeze-a-channel).
Permission can we updated through the Dashboard. You can find more information about permissions [here](https://getstream.io/chat/docs/android/chat_permission_policies/?language=kotlin)

**Reply by tristan-garaud** [reply]
This is what I did on the client side (android). My channel is flag as frozen but able to send message.
I did that:
```
val channelClient = client.channel("messaging", "general")

channelClient.updatePartial(set = mapOf("frozen" to true)).enqueue { result ->
    if (result.isSuccess) {
        val channel = result.data()
    } else {
        // Handle result.error()
    }
}
```
with the correct value ids of course.
